spec/registry.dtd

```diff
@@ -0,0 +1,33 @@
+<!ELEMENT registry (function*|pattern*)>
+
+<!ELEMENT function (description|signature+)>
+<!ATTLIST function name NMTOKEN #REQUIRED>
+
+<!ELEMENT description (#PCDATA)>
+
+<!ELEMENT pattern EMPTY>
+<!ATTLIST pattern id ID #REQUIRED>
+<!ATTLIST pattern regex CDATA #REQUIRED>
+
+<!ELEMENT signature (input*|param*|match*)>
+<!ATTLIST signature type (match|format) #REQUIRED>
+<!ATTLIST signature position (open|close|standalone) "standalone">
+<!ATTLIST signature locales NMTOKENS #IMPLIED>
+
+<!ELEMENT input EMPTY>
+<!ATTLIST input title CDATA #IMPLIED>
+<!ATTLIST input values NMTOKENS #IMPLIED>
+<!ATTLIST input pattern NMTOKEN #IMPLIED>
+
+<!ELEMENT param EMPTY>
+<!ATTLIST param title CDATA #IMPLIED>
+<!ATTLIST param name NMTOKEN #REQUIRED>
+<!ATTLIST param values NMTOKENS #IMPLIED>
+<!ATTLIST param default NMTOKEN #IMPLIED>
+<!ATTLIST param pattern IDREF #IMPLIED>
+<!ATTLIST param required (true|false) "false">
+<!ATTLIST param readonly (true|false) "false">
```

Comment:
```suggestion
<!ATTLIST param translate (true|false) "true">
```

Given that developers and translators will both be working with MF2, it might be clearer to use `translate` rather than `readonly` to communicate the intent of this attribute.

As a related question, shouldn't this also be available for `<input>`?

Reply:
In general, I'm not fond of `true` being the default value. I think I'd prefer to name these attributes such that their lack is equal to `false`.

Reply:
Re. `readonly`/`translate` for `<input>` -- good point. I think I'll drop the ability to define more than one `<input>` element per signature, then.

Reply:
> Re. `readonly`/`translate` for `<input>` -- good point. I think I'll drop the ability to define more than one `<input>` element per signature, then.

Done in 831a9cdaca37aa26d6299ae1680d3ea134f33e99.